contracts/interfaces/IUniswapV3SwapRouter.sol

```diff
@@ -0,0 +1,52 @@
+pragma solidity 0.5.17;
+pragma experimental ABIEncoderV2;
+
+interface IUniswapV3SwapRouter {
```

Comment:
is this full univ3 interface? if not please add it all since probably we're going to reuse it in other places